packages/@aws-cdk/aws-route53/test/util.test.ts

```diff
@@ -71,4 +72,24 @@ describe('util', () => {
     // THEN
     expect(qualified).toEqual('test.domain.com.');
   });
+
+  test('determineFullyQualifiedDomainName should not end with \'.\' when hostedZone.zoneName is an unresolved token', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    // WHEN
+    const providedName = 'test';
+    new cdk.CfnOutput(stack, 'zoneNameExample', {
```

Comment:
You shouldn't actually need to create a `CfnOutput` for the `importValue` to reference for this test to succeed